### PR TITLE
added Role, CustomUserDetailsService, InjectController

### PR DESCRIPTION
**Pull request by Baldinini**

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/mate/academy/controllers/InjectController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.Set;
+import javax.annotation.PostConstruct;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.service.RoleService;
+import mate.academy.service.service.UserService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Controller;
+
+@Controller
```
Comment (Sofasmile) [inline src/main/java/mate/academy/controllers/InjectController.java]: better use Component annotation here

File: src/main/java/mate/academy/model/Role.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        private final String roleName;
+
+        private RoleName(String roleName) {
+            this.roleName = roleName;
+        }
+
+        public String toString() {
+            return this.roleName;
+        }
```
Comment (Sofasmile) [inline src/main/java/mate/academy/model/Role.java]: do we use this code?

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -1,28 +1,40 @@
...
         this.shoppingCartService = shoppingCartService;
+        this.encoder = encoder;
+        this.roleService = roleService;
     }
 
     @Override
     public User register(String email, String password) {
         User user = new User();
-        user.setPassword(password);
+        user.setPassword(encoder.encode(password));
```
Comment (Sofasmile) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java]: image

File: src/main/java/mate/academy/controllers/InjectController.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+public class InjectController {
```
Comment (Sofasmile) [inline src/main/java/mate/academy/controllers/InjectController.java]: better rename this class to DataInitializer

File: src/main/java/mate/academy/security/CustomUserDetailsService.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            User user = userOptional.get();
+            UserBuilder builder = withUsername(user.getEmail());
+            builder.password(user.getPassword());
+            builder.roles(user.getRoles()
+                    .stream()
+                    .map(r -> r.getRoleName().toString())
+                    .toArray(String[]::new));
+            return builder.build();
+        }
+        throw new UsernameNotFoundException("User not found");
```
Comment (Sofasmile) [inline src/main/java/mate/academy/security/CustomUserDetailsService.java]:
```suggestion
        throw new UsernameNotFoundException("User not found: " + email);
```

File: src/main/java/mate/academy/config/SecurityConfig.java [hunk 6]
```diff
@@ -2,26 +2,38 @@
...
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers("/register").permitAll()
+                .antMatchers("/shopping-carts/**", "/orders/**").authenticated()
+                .antMatchers(HttpMethod.POST).hasRole("ADMIN")
+                .antMatchers(HttpMethod.DELETE).hasRole("ADMIN")
+                .antMatchers(HttpMethod.PUT).hasRole("ADMIN")
```
Comment (Sofasmile) [inline src/main/java/mate/academy/config/SecurityConfig.java:36]: does it work correctly?

File: src/main/java/mate/academy/model/User.java [hunk 7]
```diff
@@ -16,6 +19,8 @@
     @Column(unique = true)
     private String email;
     private String password;
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (Sofasmile) [inline src/main/java/mate/academy/model/User.java]: is FetchType.EAGER good practice?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/controllers/DataInitializer.java [hunk 8]
```diff
@@ -7,16 +7,16 @@
...
-import org.springframework.stereotype.Controller;
+import org.springframework.stereotype.Component;
 
-@Controller
-public class InjectController {
+@Component
+public class DataInitializer {
     private final RoleService roleService;
     private final UserService userService;
     private final PasswordEncoder encoder;
```
Comment (Sofasmile) [inline src/main/java/mate/academy/controllers/DataInitializer.java]: let's remove PasswordEncoder from here

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**